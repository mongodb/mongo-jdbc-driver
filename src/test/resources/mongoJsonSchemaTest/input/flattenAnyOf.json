{
  "bsonType": "object",
  "anyOf": [
    {"bsonType": "double"},
    {
      "bsonType": "date"
    },
    {
      "anyOf": [
        {"bsonType": "regex"},
        {
          "anyOf": [
            {
              "bsonType": "symbol"
            },
            {
              "bsonType": "objectId"
            },
            {
              "bsonType": "decimal"
            }
          ]
        }
      ]
    }
  ],
  "properties": {
    "bar": {
      "anyOf": [
        {"bsonType": "null"},
        {
          "bsonType": "bool",
          "additionalProperties": false
        },
        {
          "anyOf": [
            {"bsonType": "int"},
            {
              "anyOf": [
                {
                  "bsonType": "null"
                },
                {
                  "bsonType": "bool",
                  "additionalProperties": true
                },
                {
                  "bsonType": "string"
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
