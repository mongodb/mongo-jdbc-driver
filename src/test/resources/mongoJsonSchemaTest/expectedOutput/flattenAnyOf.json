{
  "bsonType": "object",
  "anyOf": [
    {
      "bsonType": "symbol"
    },
    {
      "bsonType": "decimal"
    },
    {
      "bsonType": "regex"
    },
    {
      "bsonType": "objectId"
    },
    {
      "bsonType": "date"
    },
    {
      "bsonType": "double"
    }
  ],
  "properties": {
    "bar": {
      "anyOf": [
        {"bsonType": "null"},
        {
          "bsonType": "bool",
          "additionalProperties": false
        },
        {"bsonType": "int"},
        {
          "bsonType": "bool",
          "additionalProperties": true
        },
        {
          "bsonType": "string"
        }
      ]
    }
  }
}
